File: packages/@juspay-tech/react-native-hyperswitch/android/src/main/java/io/hyperswitch/paymentsession/WalletSessionHandler.kt
package io.hyperswitch.paymentsession

import io.hyperswitch.paymentsheet.PaymentResult

interface WalletSessionHandler {
    fun isWalletEligible(wallet: String): Boolean
    fun getEligibleWallets(): List<String>
    fun launchWallet(wallet: String, resultHandler: (PaymentResult) -> Unit)
}

File: packages/@juspay-tech/react-native-hyperswitch/ios/hyperswitchSDK/Core/HyperSession/WalletSessionHandler.swift
import Foundation

public struct WalletSessionHandler {
    public let isWalletEligible: (_ wallet: String) -> Bool
    public let getEligibleWallets: () -> [String]
    public let launchWallet: (_ wallet: String, _ resultHandler: @escaping (PaymentResult) -> Void) -> Void

    public init(
        isWalletEligible: @escaping (_ wallet: String) -> Bool,
        getEligibleWallets: @escaping () -> [String],
        launchWallet: @escaping (_ wallet: String, _ resultHandler: @escaping (PaymentResult) -> Void) -> Void
    ) {
        self.isWalletEligible = isWalletEligible
        self.getEligibleWallets = getEligibleWallets
        self.launchWallet = launchWallet
    }
}
